fsspec/fuse.py

```diff
@@ -164,3 +213,122 @@ def run(fs, path, mount_point, foreground=True, threads=False):
             pass
+
+
+def mount(
+    url,
+    source_path,
+    mount_point,
+    log_file,
+    option,
+    foreground=True,
+    threads=False,
+    ready_file=False,
+):
+    kwargs = {}
+    for item in option:
+        key, value = item.split("=", 1)
+        if value.lower().endswith("[int]"):
+            value = int(value[: -len("[int]")])
+        elif value.lower().endswith("[bool]"):
+            value = value[: -len("[bool]")].lower() in ["1", "yes", "true"]
+
+        if "-" in key:
+            fs_name, setting_name = key.split("-", 1)
+            if fs_name in kwargs:
+                kwargs[fs_name][setting_name] = value
+            else:
+                kwargs[fs_name] = {setting_name: value}
+        else:
+            kwargs[key] = value
+
+    fs, url_path = url_to_fs(url, **kwargs)
+    logger.debug("Mounting %s to %s", url_path, mount_point)
+    run(fs, source_path, mount_point, log_filename=log_file, ready_file=ready_file)
+
+
+def main(args):
+    """Mount filesystem from chained URL to MOUNT_POINT.
+
+    Examples:
+
+    python3 -m fsspec.fuse memory /usr/share /tmp/mem
+
+    python3 -m fsspec.fuse local /tmp/source /tmp/local \\
+            -l /tmp/fsspecfuse.log
+
+    You can also mount chained-URLs and use special settings:
+
+    python3 -m fsspec.fuse 'filecache::zip::file://data.zip' \\
+            / /tmp/zip \\
+            -o 'filecache-cache_storage=/tmp/simplecache'
+
+    You can specify the type of the setting by using `[int]` or `[bool]`:
+
+    python3 -m fsspec.fuse 'simplecache::ftp://ftp1.at.proftpd.org' \\
+            /historic/packages/RPMS /tmp/ftp \\
+            -o 'simplecache-cache_storage=/tmp/simplecache' \\
+            -o 'simplecache-check_files=false[bool]' \\
+            -o 'ftp-listings_expiry_time=60[int]' \\
+            -o 'ftp-username=anonymous' \\
+            -o 'ftp-password=xieyanbo'
+    """
+
+    class RawDescriptionArgumentParser(argparse.ArgumentParser):
+        def format_help(self):
+            usage = super(RawDescriptionArgumentParser, self).format_help()
+            parts = usage.split("\n\n")
+            parts[1] = self.description.rstrip()
+            return "\n\n".join(parts)
+
+    parser = RawDescriptionArgumentParser(prog="fsspec.fuse", description=main.__doc__)
+    parser.add_argument(
+        "--version", action="version", version=get_versions()["version"]
+    )
+    parser.add_argument("url", type=str, help="fs url")
+    parser.add_argument("source_path", type=str, help="source directory in fs")
+    parser.add_argument("mount_point", type=pathlib.Path, help="local directory")
+    parser.add_argument(
+        "-o",
+        "--option",
+        action="append",
+        help="Any options of protocol included in the chained URL",
+    )
+    parser.add_argument(
+        "-l", "--log-file", type=pathlib.Path, help="Enable FUSE debug logging"
+    )
+    parser.add_argument(
+        "-f",
+        "--foreground",
+        default=True,
```

Comment:
This suggests the default is `True`, but the comment below says the default is `False`. Is this a typo or the underlying logic handles it that way? (I think it makes sense to just mark this argument with `action="store_false"`)

Reply:
Good catch, thanks!